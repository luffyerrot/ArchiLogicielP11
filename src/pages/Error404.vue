<template>
    <v-card class="mx-auto px-6 py-8 mt-4 message" max-width="800">
        <h1>La page n'existe pas !</h1>
        <h2>Erreur 404</h2>
    </v-card>
</template>

<style scoped lang="scss">
  .message {
    text-align: center;
  }
</style>