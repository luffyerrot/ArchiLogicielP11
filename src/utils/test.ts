import { ComponentMountingOptions, mount, shallowMount, VueWrapper } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { expect, it } from "vitest";
import { createVuetify } from "vuetify";
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

export function factory(originalComponent: any, shallow = false, pinia = false, options?: ComponentMountingOptions<any, unknown>): VueWrapper<unknown, any> {
    if (pinia) {
        setActivePinia(createPinia());
    }
    if (shallow) {
        return shallowMount(originalComponent, options);
    } else {
        const vuetify = createVuetify({
            components,
            directives,
        });
        const defaultOptions = {
            global: {
                plugins: [vuetify]
            },
            ...options
        };
        return mount(originalComponent, defaultOptions);
    }
};

export function renderCorrectly(wrapper: VueWrapper<unknown, any>) {
    it('Bon affichage de la page', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
}