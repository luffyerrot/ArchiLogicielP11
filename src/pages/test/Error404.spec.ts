import { factory, renderCorrectly } from "@/utils/test";
import { describe } from "vitest";
import Error404 from "../Error404.vue";

describe('Page 404', () => {
    const wrapper = factory(Error404, true);
    renderCorrectly(wrapper);
})