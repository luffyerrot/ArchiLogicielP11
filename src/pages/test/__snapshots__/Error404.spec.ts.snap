// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Page 404 > Bon affichage de la page 1`] = `"<v-card-stub data-v-d27eadc0="" class="mx-auto px-6 py-8 mt-4 message" maxwidth="800" disabled="false" flat="false" hover="false" ripple="true" border="false" density="default" loading="false" tile="false" replace="false" exact="false" tag="div" variant="elevated"></v-card-stub>"`;
